app/services/handover_date_service.rb

```diff
@@ -1,15 +1,16 @@
 # frozen_string_literal: true
 
 class HandoverDateService
-  Responsibility = Struct.new(:description, :custody?, :case_owner) do
+  Responsibility = Struct.new(:description, :custody?, :case_owner, :community?) do
     def to_s
       description
     end
   end
 
-  RESPONSIBLE = Responsibility.new 'Responsible', true, 'Custody'
-  SUPPORTING = Responsibility.new 'Supporting', false, 'Community'
-  UNKNOWN = Responsibility.new 'Unknown', false, 'Unknown'
+  RESPONSIBLE = Responsibility.new 'Responsible', true, 'Custody', false
+  SUPPORTING = Responsibility.new 'Supporting', false, 'Community', true
+  # 'Null Object' pattern
+  UNKNOWN = Responsibility.new 'Unknown', false, 'Unknown', false
```

Comment:
It would be nice & a bit cleaner if we could refactor this service to also include COM responsibility as part of the response. That way we wouldn't have to call the `.com_supporting?` method directly in `OffenderBase#needs_com?`.

It looks like we're half way there – the `Responsibility` object now has a `.community?` method. But it looks like the values are hard-coded based on whether the POM is `RESPONSIBLE` or `SUPPORTING` rather than being the result of the calculation in `.com_supporting?`.

It would be great if `.com_supporting?` could be a private method, so our only public interface continues to be `.handover` and `. calculate_pom_responsibility`.

Reply:
That's funny - half of me was thinking that very thing

Reply:
We've just talked through some ideas and came up with:

- HandoverData should be the single returned object which combines handover dates and responsibility (i.e. remove Responsibility sub-object and promote it to be top-level)
- Replace: HandoverData.responsibility.custody? / .community?
- With: HandoverData.community.responsible? / .supporting? AND .custody.responsible? / .supporting?
- This new model supports tri-state community involvement (not involved, supporting, responsible)
- Avoid passing back strings / descriptions whatsoever – move these to a view helper (where they belong)
- Don't worry about Law of Demeter :-) this is two booleans, so simple enough to not be a problem
- 'Unknown' responsibility == neither custody nor community are responsible (all answers are false) – this is a 'null object' for working around bad data which has made its way into the service